app/services/AtsListService.scala

```diff
@@ -101,16 +102,18 @@ class AtsListService @Inject()(
     }
 
     // TODO: Audit Events
-    for (data <- gotData) yield {
-      data.errors match {
-        case None =>
-          sendAuditEvent(account, data)
-          storeAtsListData(data)
-        case Some(IncomingAtsError("NoAtsError")) => storeAtsListData(data)
-        case Some(_)                              => Future(data)
-      }
+    gotData flatMap {
+      case AtsSuccessResponseWithPayload(payload: AtsListData) =>
+        for {
+          _    <- sendAuditEvent(account, payload)
+          data <- storeAtsListData(payload)
```

Comment:
Looks like this is the wrong way round, we should do stuff first then audit on success